x.c

```diff
@@ -1665,8 +1718,357 @@ xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len, int x, i
 
 	/* Render underline and strikethrough. */
 	if (base.mode & ATTR_UNDERLINE) {
-		XftDrawRect(xw.draw, fg, winx, winy + win.cyo + dc.font.ascent + 1,
- 				width, 1);
+		// Underline Color
```

Comment:
i believe whole this chunk is better to extract into separate function or even file

Reply:
I've kept only one method to render undercurls so this should no longer be an issue, it also means xrdb support is no longer needed

Reply:
i'd still would wait for his feedback on this point before merging

Reply:
> i believe whole this chunk is better to extract into separate function or even file (but need to have @neeasade 's input on if we are trying to merge the st patches 'as is' or with reasonable codestyle improvements)

Prefering 'as is' when possible/separate files are better when feasible

Reply:
@neeasade the patch includes 3 directive `#if` caluses controlled by a `#define` variable, 2 of these caluses render the undercurl in non-standard (and in a way that covers characters) methods, what I did was remove those and use the single clause which is now a normal `if` statement (check the latest commit)

I've also tried abstracting it into its own separate file but it's not doable since it's using data types defined within x.c itself (`XWindow` for example)

Reply:
Yes, that makes sense, thanks!. I think @actionless's suggestion to shove this into it's own function at least still would still be an improvement, as it's a large chunk dedicated to squiglies

Reply:
sounds good. on a side note, the original patch does stuff like heavy commenting (even drawing ascii stuff) and breaking simple lines into multiple just for the sake of commenting them, take a look at this for example
![image](https://user-images.githubusercontent.com/16624558/144330506-db40f50b-628f-4bc2-ba18-a8fd7ad79b19.png)

does all that go into the extracted function?

Reply:
it's a bit funky but not criminal :)

Reply:
> does all that go into the extracted function?

yes, I think that's fine/sounds good to me. I'm eyeing [this chunkish](https://github.com/gnotclub/xst/pull/137/files#diff-847ba343b012c9d0a16320307daf487df677d6945af4bdb1fb2b762f02def4daR1723-R1921)

Reply:
![2021-12-02--1638403076_1595x625_scrot](https://user-images.githubusercontent.com/1655669/144333268-88f5b81a-0c6e-4ef1-8b62-0ec63659233d.png)